kotlin-inject-compiler/test/src/test/kotlin/me/tatarka/inject/test/RoundsTest.kt

```diff
@@ -263,4 +263,57 @@ class RoundsTest {
             doesNotContain("Cannot provide", "as it is already provided")
         }
     }
+
+    @Test
+    fun can_generate_for_inject_typealias_on_primary_constructor() {
```

Comment:
This class is for running the process in multiple rounds. You should add an example to the [integration tests](https://github.com/evant/kotlin-inject/tree/main/integration-tests/common/src/test/kotlin/me/tatarka/inject/test) instead. You'll find the majority of tests are in there.

Reply:
I moved the tests to separate file in integration tests module, and added two more tests. I will try to make type-aliasing `@Component` work in a separate pull requests, so this one will not grow more.